### PR TITLE
Fixed compile errors on macOS Catalina

### PR DESCRIPTION
**Pull request by ernestp**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Build/Testing/Packaging Improvement

Short description (up to few sentences):
Fixed compile errors on macOS Catalina
...

Detailed description (optional):

...

### CODE REVIEW

File: contrib/arrow-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -70,6 +70,10 @@ add_custom_command(OUTPUT orc_proto.pb.h orc_proto.pb.cc
         --cpp_out="${CMAKE_CURRENT_BINARY_DIR}"
         "${PROTO_DIR}/orc_proto.proto")
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
```
Comment (alexey-milovidov) [inline contrib/arrow-cmake/CMakeLists.txt]: Why do we need this?
(you may add a comment)

Comment (ernestp) [reply contrib/arrow-cmake/CMakeLists.txt]: @alexey-milovidov arrow-cmake cmake file calling orc cmake subroutine which detects certain compiler features. Apple Clang compiler failed to compile this code without specifying c++11 subset. As result these compiler features detected as absent. In result it results in fail of compile orc itself.  In orc makefile there is code that sets flags, but arrow-cmake ignores these flags.

Comment (alexey-milovidov) [reply contrib/arrow-cmake/CMakeLists.txt]: Ok. I've copy-pasted your explanation into code.

**Review by alexey-milovidov — APPROVED** [review]

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
It breaks OS X build with cross-compilation:
https://clickhouse-builds.s3.yandex.net/7585/de1e5d0e533f280b93985ff54ea90b238ebb061d/build_log_540212353_1572631230.txt

`error: unknown warning option '-Wno-ctad-maybe-unsupported' [-Werror,-Wunknown-warning-option]`

**Comment by ernestp** [comment]
With cross compilation there is two points:

1. Cross compilation uses clang-8

compiler version check to include -Wno-ctad-maybe-unsupported is 'OS_DARWIN AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11'

So build should not add this option for clang-8

Can you confirm that cross compilation is working on master?

2. I cannot validate cross compilation is working on master on my setup (Ubuntu 18.0.4). Configuration step is failed.

CMake Error at /usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Threads (missing: Threads_FOUND)
Call Stack (most recent call first):
  /usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/share/cmake-3.12/Modules/FindThreads.cmake:205 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/darwin/default_libs.cmake:29 (find_package)
  CMakeLists.txt:234 (include)

-- Configuring incomplete, errors occurred!

@alexey-milovidov

File: contrib/arrow-cmake/CMakeLists.txt (hunk 1 above)
Comment (ernestp) [reply contrib/arrow-cmake/CMakeLists.txt]: Oh.. you mean to add comment to code?

**Comment by alexey-milovidov** [comment]
Yes, cross-compilation is working in master. And we ensure it with these checks.
It's easy to confirm, because we run this check for every commit:
1. Go to the list of commits: https://github.com/ClickHouse/ClickHouse/commits/master
2. Find commit with green check mark and with modifications to anything except docs
3. Click on green check mark and find "ClickHouse build check"
4. Click on "Details" link and find "darwin" build in the list.
5. There you will find links to built binaries and build logs.

File: contrib/arrow-cmake/CMakeLists.txt (hunk 1 above)
Comment (alexey-milovidov) [reply contrib/arrow-cmake/CMakeLists.txt]: Yes, otherwise comments will be lost :)

File: contrib/libcxx-cmake/CMakeLists.txt [hunk 2]
```diff
@@ -44,7 +44,7 @@ target_include_directories(cxx SYSTEM BEFORE PUBLIC $<BUILD_INTERFACE:${LIBCXX_S
 target_compile_definitions(cxx PRIVATE -D_LIBCPP_BUILDING_LIBRARY -DLIBCXX_BUILDING_LIBCXXABI)
 
 target_compile_options(cxx PUBLIC -nostdinc++ -Wno-reserved-id-macro)
-if (OS_DARWIN AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9)
+if (OS_DARWIN AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11)
```
Comment (alexey-milovidov) [inline contrib/libcxx-cmake/CMakeLists.txt]: This modification alters logic for clang 8.1.

Comment (alexey-milovidov) [reply contrib/libcxx-cmake/CMakeLists.txt]: I have edited it a little, let's see...

**Comment by alexey-milovidov** [comment]
This is the instruction on how to run cross-compilation by yourself:
https://clickhouse.yandex/docs/en/development/build_cross/

**Comment by ernestp** [comment]
I used this link and build failed:

```
CMake Error at /usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
Could NOT find Threads (missing: Threads_FOUND)
Call Stack (most recent call first):
/usr/local/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
/usr/local/share/cmake-3.12/Modules/FindThreads.cmake:205 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
cmake/darwin/default_libs.cmake:29 (find_package)
CMakeLists.txt:234 (include)

-- Configuring incomplete, errors occurred!
```

**Comment by alexey-milovidov** [comment]
It looks like SDK (system headers and libraries) is not installed... but I will ask @abyss7 (he is the author of cross-compilation) and he will help.

**Comment by ernestp** [comment]
I checked that SDK, thread library and header is in place. Must be some issue with paths.

**Comment by alexey-milovidov** [comment]
ClickHouse build check — 15/15 builds are OK

**Pull request merged**